src/aprl/envs/multi_agent.py

```diff
@@ -288,11 +288,89 @@ def _tuple_pop(input, i):
     return tuple(output), elt
 
 
+def _tuple_replace(input, i, obj):
+    output = list(input)
+    del output[i]
+    output.insert(i, obj)
+    return tuple(output)
+
+
 def _tuple_space_filter(tuple_space, filter_idx):
     filtered_spaces = (space for i, space in enumerate(tuple_space.spaces) if i != filter_idx)
     return gym.spaces.Tuple(tuple(filtered_spaces))
 
 
+def _tuple_space_replace(tuple_space, replace_idx, replace_space):
+    current_spaces = tuple_space.spaces
+    old_space = current_spaces.pop(replace_idx)
+    if type(old_space) != type(replace_space):
+        raise TypeError("Replacement action space has different type than original")
+    current_spaces.insert(replace_idx, replace_space)
+    return gym.spaces.Tuple(tuple(current_spaces))
+
+
+def _tuple_space_augment(tuple_space, augment_idx, augment_space):
+    current_space = tuple_space.spaces[augment_idx]
+    new_low = np.concatenate([current_space.low, augment_space.low])
+    new_high = np.concatenate([current_space.high, augment_space.high])
+
+    new_space = gym.spaces.Box(low=new_low, high=new_high)
+    return _tuple_space_replace(tuple_space, augment_idx, new_space)
+
+
+class MergeAgentVecEnv(VecMultiWrapper):
+    """Allows merging of two agents into a pseudo-agent by merging their actions"""
+    def __init__(self, venv, policy, replace_action_space, merge_agent_idx):
+        """Expands one of the players in a VecMultiEnv.
+        :param venv(VecMultiEnv): the environments.
+        :param policy(Policy): the fixed policy to use at merge_agent_idx
+        :param replace_action_space(Box): the action space of the new agent to merge
+        :param merge_agent_idx(int): the index of the agent that will be merged with the new agent
+        :return: a new VecMultiEnv with the same number of agents. It behaves like venv but
+                 with all actions at index merge_agent_idx merged with a fixed policy."""
+        super().__init__(venv)
+
+        assert venv.num_agents >= 1  # same as in CurryVecEnv
+        self.observation_space = _tuple_space_augment(self.observation_space, merge_agent_idx,
+                                                      self.action_space.spaces[merge_agent_idx])
+        if replace_action_space.shape != self.action_space.spaces[merge_agent_idx].shape:
+            raise ValueError("Replacement action space has different shape than original.")
+        self.action_space = _tuple_space_replace(self.action_space, merge_agent_idx,
+                                                 replace_action_space)
+
+        self._agent_to_merge = merge_agent_idx
+        self._policy = policy
+        self._action = None
+        self._state = None
+        self._obs = None
+        self._dones = [False] * venv.num_envs
+
+    def step_async(self, actions):
+        actions_copy = list(actions)
+        actions_copy[self._agent_to_merge] += self._action
```

Comment:
I think this still doesn't quite fix the bug. The problem is `list` does a shallow copy: it creates a new list, with references to the *same* objects as the original. But the issue isn't that we're changing `actions_copy`: it's that we're changing the object inside of `actions_copy`. I think it'd be OK to write:
```
new_action = actions[self._agent_to_merge] + self._action
actions = _tuple_replace(actions, self._agent_to_merge, new_action)
self.venv.step_async(actions)
```

See https://realpython.com/copying-python-objects/